psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -1719,15 +1719,15 @@ std::vector<SharedMatrix> MintsHelper::electric_field(const std::vector<double>
 SharedMatrix MintsHelper::induction_operator(SharedMatrix coords, SharedMatrix moments) {
     SharedMatrix mat = std::make_shared<Matrix>("Induction operator", basisset_->nbf(), basisset_->nbf());
     ContractOverDipolesFunctor dipfun(moments, mat);
-    auto field_integrals_ = static_cast<ElectricFieldInt *>(integral_->electric_field());
+    auto field_integrals_ = static_cast<ElectricFieldInt *>(integral_->electric_field().release());
```

Comment:
Can this be `get()` rather than `release()`? I wasn't sure where the memory management occurs otherwise eg in `compute_with_functor`

Reply:
Can this just be `auto field_integrals = integral_->electric_field()`? Looking at the code, there's no reason to bother with the static cast. The same applies below.

Reply:
Seems like the static cast is required to resolve `compute_with_functor`. I have released, cast and then put in another unique_ptr, its a bit ugly but i suppose it works. Let me know if you want it done differently.

Reply:
@jturney Why does `integral_->electric_field()` return a `OneBodyAoInt` rather than an `ElectricFieldInt`? I would assume it should return an object of the child class and leave it to the caller to decide when to upcast to a member of the parent class.

Reply:
While this should be a separate PR, `integral_->electric_field()` should return an `ElectricFieldInt` rather than a `OneBodyAOInt`. This will clean up `MintsHelper::induction_operator`. I'm not sure your version of the code handles memory correctly, but it can't be worse than the old way of doing it.

Reply:
I can raise an issue.

Reply:
raised #2793